src/internal/utils/mod.rs

```diff
@@ -1,7 +1,39 @@
 pub(crate) mod base62;
+
 pub(crate) use base62::encode as base62_encode;
+use std::path::Path;
+use std::{fs, io};
 
 #[cfg(target_os = "linux")]
 mod libc;
 #[cfg(target_os = "linux")]
 pub(crate) use libc::detect_libc;
+
+pub(crate) fn safe_rename<P: AsRef<Path>, Q: AsRef<Path>>(from: P, to: Q) -> io::Result<()> {
+    if fs::rename(from.as_ref(), to.as_ref()).is_ok() {
+        return Ok(());
+    }
```

Comment:
Can we validate the kind of error to avoid creating a copy if there is another kind of issue like permission?

Reply:
Unfortunately this error code is only checkable on nightly: https://doc.rust-lang.org/std/io/enum.ErrorKind.html#variant.CrossesDevices

I figure if it fails for perms, the copy will immediately after that, too, so not exactly a huge problem.